File: LeetCode/Easy/9-palindrome-number/try-1.java
/*
 * Problem #9: Palindrome Number
 * Difficulty: Easy
 * Submission: Try 1
 * status: Accepted
 * Language: java
 * Date: 7/8/2026, 7:31:33 PM
 * Link: https://leetcode.com/problems/palindrome-number/
 */

class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0)
            return false;
        int temp = x;
        int rev = 0;
        while (x != 0) {
            int ld = x % 10;
            rev = rev * 10 + ld;
            x = x / 10;
        }
        return temp == rev;
    }
}

File: Codeforces/71/A/try-1.py
/*
 * Problem : Way Too Long Words
 * Difficulty: Easy
 * Submission: Try 1
 * status: Accepted
 * Language: Python 3
 * Date: 7/14/2026, 12:39:43 PM
 * Link: https://codeforces.com/contest/71/problem/A
 */

n = int(input())

for _ in range(n):
    word = input()

    if len(word) <= 10:
        print(word)
    else:
        print(word[0] + str(len(word) - 2) + word[-1])
